src/sidebar/store/modules/test/real-time-updates-test.js

```diff
@@ -0,0 +1,162 @@
+      updatedAnnotations: updates,
+    });
+    return updates;
+  }
+
+  function addPendingDeletions(store) {
+    const deletions = [{ id: 'deleted-ann' }];
+    store.receiveRealTimeUpdates({
+      deletedAnnotations: deletions,
+    });
+    return deletions;
+  }
+
+  describe('receiveRealTimeUpdates', () => {
+    it("adds pending updates where the focused group matches the annotation's group", () => {
+      addPendingUpdates(store);
+      assert.deepEqual(store.pendingUpdates(), {
+        'updated-ann': { id: 'updated-ann', group: 'group-1' },
+        'created-ann': { id: 'created-ann', group: 'group-1' },
+      });
+    });
+
+    it("does not add pending updates if the focused group does not match the annotation's group", () => {
+      fakeFocusedGroupId.returns('other-group');
+      assert.deepEqual(store.pendingUpdates(), {});
+    });
+
+    it('always adds pending updates in the stream where there is no focused group', () => {
+      fakeFocusedGroupId.returns(null);
+      fakeIsSidebar.returns(false);
+
+      addPendingUpdates(store);
+
+      assert.deepEqual(store.pendingUpdates(), {
+        'updated-ann': { id: 'updated-ann', group: 'group-1' },
+        'created-ann': { id: 'created-ann', group: 'group-1' },
+      });
+    });
+
+    it('adds pending deletions if the annotation exists locally', () => {
+      fakeAnnotationExists.returns(true);
+      addPendingDeletions(store);
+      assert.deepEqual(store.pendingDeletions(), {
+        'deleted-ann': true,
+      });
+    });
+
+    it('does not add pending deletions if the annotation does not exist locally', () => {
+      fakeAnnotationExists.returns(false);
+      addPendingDeletions(store);
+      assert.deepEqual(store.pendingDeletions(), {});
+    });
+  });
+
+  describe('clearPendingUpdates', () => {
+    it('clears pending updates', () => {
+      addPendingUpdates(store);
+      store.clearPendingUpdates();
+      assert.deepEqual(store.pendingUpdates(), {});
+    });
+
+    it('clears pending deletions', () => {
+      addPendingDeletions(store);
+      store.clearPendingUpdates();
+      assert.deepEqual(store.pendingDeletions(), {});
+    });
+  });
+
+  describe('pendingUpdateCount', () => {
+    it('returns the total number of pending updates', () => {
+      const updates = addPendingUpdates(store);
+      const deletes = addPendingDeletions(store);
+      assert.deepEqual(
+        store.pendingUpdateCount(),
+        updates.length + deletes.length
+      );
+    });
+  });
+
+  it('clears pending updates when annotations are added/updated', () => {
+    const updates = addPendingUpdates(store);
+
+    // Dispatch `ADD_ANNOTATIONS` directly here rather than using
+    // the `addAnnotations` action creator because that has side effects.
+    store.dispatch({ type: 'ADD_ANNOTATIONS', annotations: updates });
+
+    assert.deepEqual(store.pendingUpdateCount(), 0);
+  });
+
+  it('clears pending updates when annotations are removed', () => {
```

Comment:
If I comment out the `delete pendingDeletions[ann.id];` line this test still passes. We might want to add some pending deletions as well here.

Reply:
Good catch. Resolved.